Sources/Internal/SecItem.swift

```diff
@@ -0,0 +1,219 @@
+//
+//  SecItem.swift
+//  Valet
+//
+//  Created by Dan Federman and Eric Muller on 9/16/17.
+//  Copyright © 2017 Square, Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+
+internal func execute<ReturnType>(in lock: NSLock, block: () -> ReturnType) -> ReturnType {
+    lock.lock()
+    defer {
+        lock.unlock()
+    }
+    return block()
+}
+
+
+internal final class SecItem {
+    
+    // MARK: Internal Enum
+    
+    internal enum DataResult<SuccessType> {
+        case success(SuccessType)
+        case error(OSStatus)
+    }
+    
+    internal enum Result {
+        case success
+        case error(OSStatus)
+    }
+    
+    // MARK: Internal Class Properties
+
+    /// Programatically grab the required prefix for the shared access group (i.e. Bundle Seed ID). The value for the kSecAttrAccessGroup key in queries for data that is shared between apps must be of the format bundleSeedID.sharedAccessGroup. For more information on the Bundle Seed ID, see https://developer.apple.com/library/ios/qa/qa1713/_index.html
+    internal static var sharedAccessGroupPrefix: String {
```

Comment:
I assume this will always return the same thing?  If so, we could just make it non-computed property so it only has to run the query once.

Reply:
It _should_ always return the same thing. In the potential error cases below (I.e the keychain throwing some kind of shit fit) it’s possible that we’d get a bad value once but not later though. It feels safer to compute every time to prevent against an intermittent failure hosing the entire Valet instance for the lifecycle of the object. But I’m also being a bit paranoid – this code _should_ always work, barring the keychain failing at a fundamental level.